src/java.base/share/classes/java/lang/invoke/VarHandles.java

```diff
@@ -338,13 +343,7 @@ static VarHandle makeMemoryAddressViewHandle(Class<?> carrier, long alignmentMas
     }
 
     private static VarHandle maybeAdapt(VarHandle target) {
-        if (!VAR_HANDLE_IDENTITY_ADAPT) return target;
-        target = filterValue(target,
-                        MethodHandles.identity(target.varType()), MethodHandles.identity(target.varType()));
-        MethodType mtype = target.accessModeType(VarHandle.AccessMode.GET).dropParameterTypes(0, 1);
-        for (int i = 0 ; i < mtype.parameterCount() ; i++) {
-            target = filterCoordinates(target, i, MethodHandles.identity(mtype.parameterType(i)));
-        }
+        /* This optimization is disabled in OpenJ9. */
```

Comment:
`VAR_HANDLE_IDENTITY_ADAPT` is a performance enhancement, and it is disabled until OpenJ9 adopts OpenJDK VarHandles. Disabling this enhancement doesn't impact basic functionality.